semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -541,6 +557,42 @@ and expr_aux env ?(void = false) e_gen =
       let args = arguments env args in
       add_instr env (mk_i (CallSpecial (Some lval, special, args)) eorig);
       mk_e (Fetch lval) (related_tok tok)
+  | G.Call
+      (* This narrowly adds some calls from objects of certain classes as default
+         propagators. For now, this is a quick fix to make rule-writing easier.
+         TODO In the future, we should only use this for classes where we perform
+         a special analysis. Other propagators should be in a configurable defaults
+         file *)
+      ( {
+          e =
+            G.DotAccess
+              ( ({
+                   e =
+                     G.N
+                       (G.Id
+                         ((id, _), { id_type = { contents = Some { t; _ } }; _ }));
+                   _;
+                 } as obj),
+                _tok,
+                G.FN
+                  (G.Id
+                    ( ((("add" | "addAll") as kind), tok),
+                      { G.id_resolved = { contents = None }; _ } )) );
+          _;
+        },
+        args )
+    when is_collections_class t ->
+      let lval =
+        (* TODO: for cases like adding to a HashMap, we will
+           want to make this key-sensitive. We can do that
+           by adding the key as an offset to the lval *)
```

Comment:
I think that if we go this way with collections, I would rather translate them already into assignments to lval+offset. Then they would be treated like arrays, and we can configure via options how that should work. At some point, we can make them index-sensitive as you say here.